### PR TITLE
Fix validation form on the /additional_code_search page.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-4154

### What?
Fix wrong error messages on the /additional_code_search page.

### CODE REVIEW

File: app/forms/additional_code_search_form.rb [hunk 1]
```diff
@@ -54,4 +56,13 @@ def additional_code_type_ids
       @additional_code_type_ids ||= AdditionalCodeType.all.map(&:additional_code_type_id).sort
     end
   end
+
+  private
+
+  def sanitize_code
```
Comment (willfish) [inline app/forms/additional_code_search_form.rb]:
```suggestion
  def sanitize_code!
```

File: app/forms/additional_code_search_form.rb [hunk 2]
```diff
@@ -14,8 +14,10 @@ class AdditionalCodeSearchForm
   attr_writer :page
 
   def validate_code
+    sanitize_code
+
     if code.present?
-      errors.add(:code, :invalid) unless code =~ /\A([A-Z]|[0-9]){4}\z/
```
Comment (willfish) [inline app/forms/additional_code_search_form.rb:18]: This was checking length and specific characters before. I don't imagine its a huge deal no longer doing the two together but this is a different validation

File: spec/forms/additional_code_search_form_spec.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+require 'spec_helper'
```
Comment (willfish) [inline spec/forms/additional_code_search_form_spec.rb:1]: You shouldn't need this because zeitwerk should be loading this lazily

**Comment by willfish** [comment]
I'd add a small check of the sanitize code method - maybe one of the examples could include some white space. Also, if we're doing this here can we also do the other forms that take a composite code value (e.g. footnote/certificate search)

File: app/forms/additional_code_search_form.rb (hunk 2 above)
Comment (alexdesi) [reply app/forms/additional_code_search_form.rb:18]: It is just redundant.
Another validation checks if the first char is a valid type. And it works better. It checks if the type is a valid possible_types (and not just generic A-Z).

**Comment by alexdesi** [comment]
Indeed, I've just checked and we got a similar error on the footnote_search page.
Since the code to fix is in different places, even if it is similar, I've create a new ticket: https://transformuk.atlassian.net/browse/HOTT-4256

**Comment by willfish** [comment]
You'll want another for certificate search.

**Review by willfish — APPROVED** [review]

**Pull request merged**